### DISCUSSION TITLE
using deepgram live transcription with Asterisk PBX

### DISCUSSION THREAD

**Discussion by aqeelabpro** [deepgram/community#562] [Feature Requests]
I have been using Deepgram for the last year and a half for my projects. and now I wanted to use Deepgram together with Asterisk PBX for real-time speech recognition. Is there a way to implement this with Asterisk PBX?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jjmaldonis** [comment]
Hey @aqeelabpro, Deepgram has an integration with Asterisk and you can find more information here:

- https://deepgram.com/learn/teleconnx-stt-asterisk-freepbx-partnership
- https://www.teleconnx.com/dg-stt/

**Comment by aqeelabpro** [comment]
I followed their guide on https://www.teleconnx.com/dg-stt/get-started.html but getting a **Connection Failed Goodbye** playback message. I think I should consult with them. so I think I should close this.

**Comment by gcareri** [comment]
Hi @aqeelabpro ,

what you’re trying to build sounds very close to what we’re working on with Agent Voice Response (AVR).
You might want to take a look at AVR — it’s open & free, designed to connect Asterisk with modern ASR (STT), LLM, and TTS providers to create real-time conversational voice bots.

In your case, we already have an integration for Deepgram STT that might fit perfectly:
👉 https://github.com/agentvoiceresponse/avr-sts-deepgram

With AVR you can:
	•	Route calls from Asterisk directly to the AI bot
	•	Use Deepgram for transcription and TTS responses
	•	Plug in an LLM (e.g. Groq, OpenAI, or others) for conversation handling
	•	Support multiple languages (including Spanish)
	•	Build flexible call flows without starting everything from zero

We’d be happy to help you set this up and adapt it to your use case.
💬 Join our community on Discord for support and guidance: https://discord.gg/MUd3y7eGVF

Regards,
gcareri

**Reply by aqeelabpro** [reply]
@gcareri I will be testing it soon, I was impatiently waiting for something like this . Thanks for informing me.

**Reply by hkjarral** [reply]
I have deepgram implemented and available for test calls in my repo, please check it out it may be helpful https://github.com/hkjarral/Asterisk-AI-Voice-Agent
